src/sentry/services/hybrid_cloud/auth/impl.py

```diff
@@ -268,6 +269,19 @@ def disable_provider(self, *, provider_id: int) -> None:
                 auth_provider.disable_scim()
             auth_provider.delete()
 
+    def update_provider(self, organization_id: int, provider: RpcAuthProvider) -> None:
```

Comment:
Would it make sense to just pass the config + provider as separate arguments rather than passing back the entire RpcAuthProvider object?

Reply:
I guess also the auth provider id, too.